distributed/node.py

```diff
@@ -164,7 +164,7 @@ def start_http_server(
                 except Exception:
                     change_port = True
                     retries_left = retries_left - 1
-                    if retries_left < 1:
+                    if retries_left < 1:  # pragma: no cover
```

Comment:
This is pretty important piece for the retry logic. Will it retry once? Will it retry forever? Will it retry for Xs?
It is rather trivial to understand what these two lines are doing but this behaviour can easily be lost in a refactor and a test should protect us. It's fine to change this behaviour but it should be done deliberately

```suggestion
                    if retries_left < 1:
```